app/src/components/docGen.js

```diff
@@ -22,21 +22,27 @@ const docGen = {
 
       // If it's not an array of multiple data items, pass it into carbone as a singular object
       const data = body.contexts.length > 1 ? body.contexts : body.contexts[0];
+
+      // TODO: there's too much response stuff down here in a component layer, figure out how
+      // better to have the asynchronous carbone render be blocking and wait for its response
+      // up in the v1/docGen.js route layer, then handle response setting there.
       carbone.render(tmpFile.name, data, function (err, result) {
         if (err) {
-          log.error(`Error during Carbone generation. Error: ${err}`);
-          throw new Error(err);
-        }
-        // write the result
-        var readStream = new stream.PassThrough();
-        readStream.end(result);
+          const errTxt = `Error during Carbone generation. Error: ${err}`;
+          log.error(errTxt);
+          response.status(500).send(errTxt);
+        } else {
+          // write the result
+          var readStream = new stream.PassThrough();
+          readStream.end(result);
 
-        response.set('Content-disposition', 'attachment; filename=test');
-        response.set('Content-Type', 'text/plain');
+          response.set('Content-disposition', 'attachment; filename=test');
+          response.set('Content-Type', 'text/plain');
 
-        readStream.pipe(response);
-        // Doc is generated at this point, remove the input file
-        tmpFile.removeCallback();
+          readStream.pipe(response);
+          // Doc is generated at this point, remove the input file
+          tmpFile.removeCallback();
```

Comment:
This is temporary, but can probably put in a `response.status(201)` just so that it matches the API spec. We can worry about how to do this more correctly in a subsequent refactor.

Reply:
Returns a 201 now